sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs

```diff
@@ -371,10 +410,10 @@ public virtual Response<AddChatParticipantsResult> AddParticipant(ChatParticipan
             {
                 return _chatThreadRestClient.AddChatParticipants(Id, new[] { participant.ToChatParticipantInternal() }, cancellationToken);
             }
-            catch (Exception ex)
+            catch (RequestFailedException requestFailedException)
```

Comment:
We need to check if there is any invalid participant, and throw the exception if there is :

        public virtual Response<AddChatParticipantsResult> AddParticipant(ChatParticipant participant, CancellationToken cancellationToken = default)
        {
            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ChatThreadClient)}.{nameof(AddParticipant)}");
            scope.Start();
            try
            {
                Response<AddChatParticipantsResult> addChatParticipantsResult = _chatThreadRestClient.AddChatParticipants(Id, new[] { participant.ToChatParticipantInternal() }, cancellationToken);
                if (addChatParticipantsResult.Value.InvalidParticipants.Count > 0)
                {
                    throw _clientDiagnostics.CreateRequestFailedException(addChatParticipantsResult.GetRawResponse());
                }
            }
            catch (Exception ex)
            {
                scope.Failed(ex);
                throw;
            }
        }

Reply:
Fixed, question: would AddParticipant still return a success result if there was no error? or it should not return anything?

Reply:
my bad, It should not return anything:
public virtual Response AddParticipant(ChatParticipant participant, CancellationToken cancellationToken = default)